aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -0,0 +1,74 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/BatchLinearAlgebra.h>
+#include <ATen/native/LinearAlgebraUtils.h>
+
+#include <TH/TH.h>  // for USE_LAPACK
+
+namespace at { namespace native {
+
+namespace {
+
+template <typename scalar_t>
+static void apply_eig(const Tensor& self, bool eigenvectors, Tensor& vals_, Tensor& vecs_, int64_t* info_ptr) {
+#ifndef USE_LAPACK
+  TORCH_CHECK(false, "Calling torch.eig on a CPU tensor requires compiling ",
+    "PyTorch with LAPACK. Please use PyTorch built with LAPACK support.");
+#else
+  char jobvr = eigenvectors ? 'V' : 'N';
+  int64_t n = self.size(-1);
+  auto self_data = self.data_ptr<scalar_t>();
+
+  auto vals_data = vals_.data_ptr<scalar_t>();
+  scalar_t* wr = vals_data;
+  scalar_t* wi = vals_data + n;
+
+  scalar_t* vecs_data = eigenvectors ? vecs_.data_ptr<scalar_t>() : nullptr;
+  int ldvr = eigenvectors ? n : 1;
+
+  if (n > 0) {
+    // call lapackEig once to get the optimal size for work data
+    scalar_t wkopt;
+    int info;
+    lapackEig<scalar_t>('N', jobvr, n, self_data, n, wr, wi,
+      nullptr, 1, vecs_data, ldvr, &wkopt, -1, &info);
+    int lwork = static_cast<int>(wkopt);
+
+    // call again to do the actual work
+    Tensor work = at::empty({lwork}, self.dtype());
+    lapackEig<scalar_t>('N', jobvr, n, self_data, n, wr, wi,
+      nullptr, 1, vecs_data, ldvr, work.data_ptr<scalar_t>(), lwork, &info);
+    *info_ptr = info;
+  }
+#endif
+}
+
+static std::tuple<Tensor, Tensor> eig_kernel_impl(const Tensor& self, bool& eigenvectors) {
+  int64_t n = self.size(-1);
+  Tensor self_ = at::empty_strided(
```

Comment:
Why does this have to be strided? Maybe add a comment here.

Reply:
Thanks for pointing out that this is unclear! I added a comment